PocketBook: Improve existing and add new device definitions

roshavagarga opened the pull request:
Proposed changes:
- [x] Added Pocketbook Basic 611 definition
- [x] Added PocketBook Basic Lux definition
- [x] Added PocketBook Touch Lux 5 definition (#6475)
- [x] Added PocketBook Color definition (#6479)
- [x] Added PocketBook Aqua definition
- [x] Added PocketBook Ultra definition
- [x] Rearranged devices based on their FW numbers.
- [x] Fixed device name comments.
- [x] Added more device settings and rearranged them across all devices.

Notes:
1. PocketBook Basic 611 runs on the v16 firmware that predates the current OS, so compatibility needs to be tested.
2. Codenames were created based on similarity. To get the true codenames, I'd need users with said devices to check a file available in all PocketBooks that lists the proper codename. Not sure how to otherwise find the proper codenames.
3. Several of the previously added devices actually cover more than 1 device that shares the same firmware, but their codenames might differ. This should be further explored.
4. None of these definitions were tested, as I don't have the devices.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/koreader/koreader/6481)
<!-- Reviewable:end -->

roshavagarga commented:
@pazos I'll fix the properties, but on that note - what do I do with Touch devices that have buttons too - should I add 'hasFewKeys' or just let it roll with the preset 'hasKeys'? Also, I still have no idea what 'hasWifiToggle' means/does.

pazos commented:
> what do I do with Touch devices that have buttons too?

Nothing, unless you're sure they only have a 5key-dpad (left, right, up, down, enter). In these cases add `hasFewKeys=yes`, but don't change `hasKeys`.

> I still have no idea what 'hasWifiToggle' means/does.

It means that devices have wifi. Don't touch it unless you're sure that a particular model doesn't have wifi.

roshavagarga commented:
New changes:
- [x] Removed `hasWifiToggle = no` from a lot of the devices, since their manuals specifically state that they have Wi-Fi capabilities.
- [x] Removed `hasFrontlight = yes` from all devices (redundant).
- [x] Removed `isTouchDevice = yes` from all devices (redundant).
- [x] Added `hasGSensor = yes` to 740, 740_2 and 1040 definitions.

@pazos Some of the touch devices have a quad-button setup, typically back, left, right, home.

Also, I'm going to be honest, I have no idea how to efficiently split this up, outside of deleting it and starting over (non-dev, using webUI).

pazos commented:
> Also, I'm going to be honest, I have no idea how to efficiently split this up, outside of deleting it and starting over (non-dev, using webUI).

Don't worry. It is totally fine. For the next time please split the code between: new code, changes in comments and sorting. :+1: 

> Added hasGSensor = yes to 740, 740_2 and 1040 definitions - manuals mention both a g-sensor and a cover sensor

Please don't. The feature is not intended for pocketbooks and needs to be implemented first.

pazos commented on frontend/device/pocketbook/device.lua:
```diff
@@ -206,156 +206,210 @@ function PocketBook:getDeviceModel()
+
+-- PocketBook Aqua (640)
+local PocketBook640 = PocketBook:new{
+    model = "PBAqua"
```
add a comma at the end

roshavagarga commented:
Got rid of `hasGSensor`, fixed the comma and I'll keep it in mind, thanks for the advice :)

Frenzie commented:
> Don't worry. It is totally fine. For the next time please split the code between: new code, changes in comments and sorting. +1

Agreed, I mistakenly thought it was a refactoring with a new device or two so I didn't notice the redundant additions.

pazos commented on frontend/device/pocketbook/device.lua:
```diff
@@ -206,156 +206,207 @@ function PocketBook:getDeviceModel()
+    codename == "PocketBook 614" then
     return PocketBook614W
+elseif codename == "PocketBook 615" or
+    codename == "PB615" then
```
same line, please

pazos commented on frontend/device/pocketbook/device.lua:
```diff
@@ -206,156 +206,207 @@ function PocketBook:getDeviceModel()
     return PocketBook613
-elseif codename == "PocketBook 614W" then
+elseif codename == "PocketBook 614W" or
+    codename == "PocketBook 614" then
```
same line, please

pazos commented on frontend/device/pocketbook/device.lua:
```diff
@@ -372,22 +423,31 @@ elseif codename == "PB626" or codename == "PB626(2)-TL3" or
     return PocketBook630
-elseif codename == "PB631" then
+elseif codename == "PB631" or
+    codename == "PocketBook 631" then
```
same line, please

pazos commented on frontend/device/pocketbook/device.lua:
```diff
@@ -206,156 +206,207 @@ function PocketBook:getDeviceModel()
 local PocketBookColorLux = PocketBook:new{
     model = "PBColorLux",
+    display_dpi = 125,
+    isAlwaysPortrait = yes,
```
That line was added without a reason?

pazos commented on frontend/device/pocketbook/device.lua:
```diff
@@ -206,156 +206,207 @@ function PocketBook:getDeviceModel()
+local PocketBook633 = PocketBook:new{
+    model = "PBColor",
+    display_dpi = 300,
+    hasColorScreen = yes,
```
Doesn't need:

```
    has3BytesWideFrameBuffer = yes,
    canUseCBB = no, -- 24bpp
```

¿?

pazos replied:
I mean: I'm not sure what I'm talking about, but the last line of the logs in https://github.com/koreader/koreader/issues/6479#issue-674324708, before the segmentation fault, is `FB: bpp          : 24`

pinging @NiLuJe (but he's on holidays, so I would say add the definition anyways and revert back later if I was utterly wrong, which is probable)

pazos replied:
I mean: I'm not sure what I'm talking about, but the last line of the logs in https://github.com/koreader/koreader/issues/6479#issue-674324708, before the segmentation fault, is `FB: bpp          : 24`

pinging @NiLuJe (but he's on holidays, so I would say add the definition anyways and revert back later if I was utterly wrong, which is probable)

Frenzie replied:
It'll need someone with the device and all, but https://github.com/koreader/koreader-base/pull/732 with its `has3BytesWideFrameBuffer = yes,` might potentially be relevant.

Frenzie replied:
See <https://goodereader.com/blog/electronic-readers/review-of-the-pocketbook-color-lux-ereader>.

roshavagarga replied:
The Color 633 is different though - https://pocketbook.ch/en-ch/catalog/color/color-ch, hence my uncertainty, but #6479's creator might be able to test it out?

pazos replied:
yep, that's the point. merge it but don't close #6479 yet. Ping OP when a nightly is available and see if that automagically fixes it or at least something in the logs changed :p

roshavagarga replied:
Don't quote me on this, but I faintly remember reading somewhere today, that the 633 has 300dpi for black & white/grayscale and 100dpi for color or something of the sort. No idea whether that's important or useful information :/

Frenzie replied:
I didn't say to add that line, just that it's something that's there if necessary. But in the heat I overlooked that @pazos already said that.

Frenzie replied:
> No idea whether that's important or useful information :/

Yes and no; I doubt it has anything to do with what we have to feed it. ;-)
